### PR TITLE
Add the header_row parameter in read_csv

### LINKED ISSUE #3817: Add header_row parameter to read_csv in the Python API
related to: #3801

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes #3817

### CODE REVIEW

File: py/server/deephaven/csv.py [hunk 1]
```diff
@@ -74,19 +85,19 @@ def read(
...
-            .allowMissingColumns(allow_missing_columns)
-            .ignoreExcessColumns(ignore_excess_columns)
-            .delimiter(ord(delimiter))
-            .quote(ord(quote))
-            .ignoreSurroundingSpaces(ignore_surrounding_spaces)
-            .trim(trim)
-            .build()
-        )
+        csv_specs = (csv_specs_builder.hasHeaderRow(not headless)
+                     .skipHeaderRows(header_row)
```
Comment (rcaudy) [inline py/server/deephaven/csv.py:89]: I would add some kind of clarification about what happens if you say headless and specify a header_row that isn't 0. Also, why headless and not header_less, or has_header defaulting to true?

Comment (jmao-denver) [reply py/server/deephaven/csv.py:89]: Not sure, I don't like neg/neg either, could be from a prototype Devin did?

File: py/server/deephaven/csv.py [hunk 2]
```diff
@@ -38,14 +39,25 @@ def read(
     Args:
         path (str): a file path or a URL string
         header (Dict[str, DType]): a dict to define the table columns with key being the name, value being the data type
-        skip_rows (long): number of data rows to skip before processing data. This is useful when you want to parse data in chunks. Defaults to 0
-        num_rows (long): max number of rows to process. This is useful when you want to parse data in chunks. Defaults to {@link Long#MAX_VALUE} 
-        allow_missing_columns (bool): whether the library should allow missing columns in the input. If this flag is set, then rows that are too short (that have fewer columns than the header row) will be interpreted as if the missing columns contained the empty string. Defaults to false.
-        ignore_excess_columns (bool): whether the library should allow excess columns in the input. If this flag is set, then rows that are too long (that have more columns than the header row) will have those excess columns dropped. Defaults to false.
+        headless (bool): whether the csv file doesn't have a header row, default is false
```
Comment (devinrsmith) [inline py/server/deephaven/csv.py:42]: You already have this param below.

File: py/server/deephaven/csv.py (hunk 1 above)
Comment (devinrsmith) [reply py/server/deephaven/csv.py:90]: I do like `has_header` more than the negation, but I don't _think_ it was something I did... might have been carried over from previous CSV stuff?

File: py/server/deephaven/csv.py [hunk 3]
```diff
@@ -38,14 +39,25 @@ def read(
...
+        num_rows (long): max number of rows to process. This is useful when you want to parse data in chunks.
+            Defaults to {@link Long#MAX_VALUE}
+        ignore_empty_lines (bool): whether to ignore empty lines, default is False
+        allow_missing_columns (bool): whether the library should allow missing columns in the input. If this flag is
+            set, then rows that are too short (that have fewer columns than the header row) will be interpreted as if
+            the missing columns contained the empty string. Defaults to false.
+        ignore_excess_columns (bool): whether the library should allow excess columns in the input. If this flag is
+            set, then rows that are too long (that have more columns than the header row) will have those excess columns
+            dropped. Defaults to false.
         headless (bool): indicates if the CSV data is headless, default is False
```
Comment (chipkent) [inline py/server/deephaven/csv.py:56]:
```suggestion
```

File: py/server/deephaven/csv.py [hunk 4]
```diff
@@ -74,19 +86,19 @@ def read(
...
+                     .numRows(num_rows)
+                     .ignoreEmptyLines(ignore_empty_lines)
+                     .allowMissingColumns(allow_missing_columns)
+                     .ignoreExcessColumns(ignore_excess_columns)
+                     .delimiter(ord(delimiter))
+                     .quote(ord(quote))
+                     .ignoreSurroundingSpaces(ignore_surrounding_spaces)
+                     .trim(trim)
+                     .build()
+                     )
```
Comment (chipkent) [inline py/server/deephaven/csv.py:101]: formatting is worse.

**Review by chipkent — CHANGES_REQUESTED** [review]

**Review by chipkent — APPROVED** [review]

**Pull request merged**